fuzz.py

```diff
@@ -46,6 +47,16 @@ def test_idempotent_any_syntatically_valid_python(
         # able to cope with it.  See issues #970, #1012, #1358, and #1557.
         # TODO: remove this try-except block when issues are resolved.
         return
+    except TokenError as e:
+        # The Python3.7 'compile' method accepts input that terminates with a
+        # multi-line continuation character (\), but black rejects it.
+        # TODO: remove this special-case once py37 reaches end-of-life
+        exception_message = e.args[0]
+        if (
+            src_contents.endswith("\\")
```

Comment:
I don't think this condition (`endswith.("\\")`) is perfect.

This build would have still failed with that condition in place: https://github.com/psf/black/runs/1904676745

Reply:
As would https://github.com/psf/black/runs/1898084374, which is in fact a Python 3.8 build.